src/main/java/uk/gov/ons/ctp/response/action/service/ProcessEventService.java

```diff
@@ -0,0 +1,766 @@
+              SurveyDTO survey = getSurvey(c.getSurveyId().toString());
+              ActionTemplate actionTemplate = actionTemplateRepository.findByType(c.getType());
+              processEmailCase(actionCase, collectionExercise, survey, actionTemplate, c, instant);
+            });
+  }
+
+  /**
+   * Processes Email Cases. Get CaseParty against the email case If BusinessNotification populate
+   * email data and process it for each 'ACTIVE' respondentParty else to the Party
+   *
+   * @param actionCase
+   * @param collectionExercise
+   * @param survey
+   * @param actionTemplate
+   */
+  public void processEmailCase(
+      ActionCase actionCase,
+      CollectionExerciseDTO collectionExercise,
+      SurveyDTO survey,
+      ActionTemplate actionTemplate,
+      ActionEvent actionEvent,
+      Instant instant) {
+    UUID actionCaseId = actionCase.getId();
+    String templateType = actionTemplate.getType();
+    ActionTemplateDTO.Handler templateHandler = actionTemplate.getHandler();
+    log.with("caseId", actionCaseId)
+        .with("actionTemplate", templateType)
+        .with("actionHandler", templateHandler)
+        .info("Processing Email Event.");
+    boolean isSuccess = true;
+    try {
+      log.with("caseId", actionCaseId).info("Getting ActionCaseParty");
+      ActionCaseParty actionCaseParty = setParties(actionCase, survey);
+      if (isBusinessNotification(actionCase)) {
+        log.with("caseId", actionCase).info("Processing Email for isBusinessNotification true");
+        actionCaseParty
+            .getChildParties()
+            .parallelStream()
+            // .filter(respondentParty -> isActive(respondentParty))
+            .forEach(
+                respondentParty ->
+                    processEmail(
+                        actionCaseParty.getParentParty(),
+                        respondentParty,
+                        survey,
+                        actionTemplate,
+                        actionCase,
+                        collectionExercise));
+      } else {
+        log.with("caseId", actionCase).info("Processing Email for isBusinessNotification false");
+        processEmail(
+            actionCaseParty.getParentParty(),
+            actionCaseParty.getChildParties().get(0),
+            survey,
+            actionTemplate,
+            actionCase,
+            collectionExercise);
+      }
+    } catch (Exception e) {
+      log.with("caseId", actionCaseId)
+          .with("actionTemplate", templateType)
+          .with("actionHandler", templateHandler)
+          .warn("Processing Email Event FAILED.");
+      isSuccess = false;
+    }
+    if (actionEvent == null) {
+      createCaseActionEvent(
+          actionCaseId,
+          templateType,
+          templateHandler,
+          isSuccess,
+          collectionExercise.getId(),
+          survey.getId(),
+          instant);
+    } else {
+      updateCaseActionEvent(actionEvent, instant);
+    }
+  }
+
```

Comment:
It feels like a lot of the methods below belong in other classes, eg setParties feels like it should be part of ActionCaseParty or the party svc client. This is a stylistic choice, but it's worth having a think about where they most logically fit.

Reply:
Sure, you are absolutely right. This does require further refactoring. Can work on this for sure